pkg/certificate/providers/tresor/certificate_manager.go

```diff
@@ -52,29 +54,34 @@ func (cm *CertManager) issue(cn certificate.CommonName, validityPeriod time.Dura
 
 	x509Root, err := certificate.DecodePEMCertificate(cm.ca.GetCertificateChain())
 	if err != nil {
-		log.Error().Err(err).Msg("Error decoding Root Certificate's PEM")
+		log.Error().Err(err).Str(errcode.Kind, errcode.ErrDecodingPEMCert.String()).
+			Msg("Error decoding Root Certificate's PEM")
 	}
 
 	rsaKeyRoot, err := certificate.DecodePEMPrivateKey(cm.ca.GetPrivateKey())
 	if err != nil {
-		log.Error().Err(err).Msg("Error decoding Root Certificate's Private Key PEM ")
+		log.Error().Err(err).Str(errcode.Kind, errcode.ErrDecodingPEMPrivateKey.String()).
+			Msg("Error decoding Root Certificate's Private Key PEM ")
 	}
 
 	derBytes, err := x509.CreateCertificate(rand.Reader, &template, x509Root, &certPrivKey.PublicKey, rsaKeyRoot)
 	if err != nil {
-		log.Error().Err(err).Msgf("Error issuing x509.CreateCertificate command for SerialNumber=%s", serialNumber)
+		log.Error().Err(err).Str(errcode.Kind, errcode.ErrCreatingCertToIssue.String()).
```

Comment:
```suggestion
		log.Error().Err(err).Str(errcode.Kind, errcode.ErrCreatingCert.String()).
```

Same suggestion for other usages of ErrCreatingCertToIssue.